File: python/series/series.py
def slices(series, size):
    if not size > 0:
        raise ValueError("Your size argument must be an integer greater "
                         "than 0! ;_;")
    if size > len(series):
        raise ValueError("Your size argument cannot be larger than the "
                         "series' length. Sneak!")

    nums = [int(n) for n in series]

    return [nums[i:i+size]
            for i in xrange(len(nums)+1 - size)]

File: ruby/series/series.rb
class Series
  def initialize(sequence)
    @sequence = sequence
  end

  def slices(slice_len)
    if slice_len > @sequence.length
      raise ArgumentError.new("Slice length cannot be larger than sequence.")
    end
    slices = []
    num_slices = @sequence.length - slice_len + 1

    (0...num_slices).each do |i|
      slice = @sequence[i, slice_len]
      slice_ary = slice.split('').inject([]) { |m, ltr| m << ltr.to_i }
      slices << slice_ary
    end

    slices
  end
end
